### PR TITLE
Remove mesh lib on file io dependency

### PR DESCRIPTION
**Pull request by endJunction**
Somehow this dependency slipped past my mind.
Moving reading of the file into DE.

Use instead layered mesh constructor taking a GeoLib::Raster.
Reading of the raster happens inside the application now.

Typo was "wher". No changes to the comment otherwise.

### CODE REVIEW

File: Applications/DataExplorer/DataView/MshView.cpp [hunk 1]
```diff
@@ -164,7 +165,17 @@ void MshView::openMap2dMeshDialog()
...
 	if (dlg.useRasterMapping())
 	{
-		if (!MeshLib::MeshLayerMapper::layerMapping(*result, dlg.getRasterPath(), dlg.getNoDataReplacement()))
+		std::unique_ptr<GeoLib::Raster> raster{
+		    FileIO::AsciiRasterInterface::readRaster(dlg.getRasterPath())};
+		if (!raster)
+		{
+			OGSError::box(QString::fromStdString(
+			    "Error mapping mesh. Could not read raster file " +
+			    dlg.getRasterPath()));
```
Comment (TomFischer) [inline Applications/DataExplorer/DataView/MshView.cpp]: This is a nice side effect: A possible error message can be displayed directly in the DataExplorer.

**Comment by TomFischer** [comment]
:fast_forward:

**Comment by chleh** [comment]
:fast_forward: 
@endJunction, once you have finished disentangling the various OGS6 libs, could you please provide a drawing (maybe hand-drawn) depicting the dependency graph among the libs, because sometimes there could be two options: E.g., either MeshLib depends on FileIO or the other way around.

**Comment by endJunction** [comment]
For uptodate relations see Doxygen, _e.g._ for [MeshLib](https://doxygen.opengeosys.org/dir_cbf36d127bae6c5ddda0a904f4deb7f8.html) where there is MeshLib<->FileIO dependency, but will be FileIO depends on MeshLib only after this is merged.

**Pull request merged**